atomic_reactor/plugins/build_source_container.py

```diff
@@ -51,35 +51,46 @@ def run(self):
         fetch_sources_result = self.workflow.prebuild_results.get(PLUGIN_FETCH_SOURCES_KEY, {})
         source_data_dir = fetch_sources_result.get('image_sources_dir')
         remote_source_data_dir = fetch_sources_result.get('remote_sources_dir')
+        maven_source_data_dir = fetch_sources_result.get('maven_sources_dir')
 
         source_exists = source_data_dir and os.path.isdir(source_data_dir)
         remote_source_exists = remote_source_data_dir and os.path.isdir(remote_source_data_dir)
+        maven_source_exists = maven_source_data_dir and os.path.isdir(maven_source_data_dir)
 
-        if not source_exists and not remote_source_exists:
+        if not any([source_exists, remote_source_exists, maven_source_exists]):
             err_msg = "No SRPMs directory '{}' available".format(source_data_dir)
             err_msg += "\nNo Remote source directory '{}' available".format(remote_source_data_dir)
+            err_msg += "\nNo Maven source directory '{}' available".format(maven_source_data_dir)
             self.log.error(err_msg)
             return BuildResult(logs=err_msg, fail_reason=err_msg)
 
         if source_exists and not os.listdir(source_data_dir):
             self.log.warning("SRPMs directory '%s' is empty", source_data_dir)
         if remote_source_exists and not os.listdir(remote_source_data_dir):
             self.log.warning("Remote source directory '%s' is empty", remote_source_data_dir)
+        if maven_source_exists and not os.listdir(maven_source_data_dir):
+            self.log.warning("Maven source directory '%s' is empty", maven_source_data_dir)
 
         image_output_dir = tempfile.mkdtemp()
         cmd = ['bsi', '-d']
-        drivers = []
+        drivers = set()
 
         if source_exists:
-            drivers.append('sourcedriver_rpm_dir')
+            drivers.add('sourcedriver_rpm_dir')
             cmd.append('-s')
             cmd.append('{}'.format(source_data_dir))
 
         if remote_source_exists:
-            drivers.append('sourcedriver_extra_src_dir')
+            drivers.add('sourcedriver_extra_src_dir')
             cmd.append('-e')
             cmd.append('{}'.format(remote_source_data_dir))
 
+        if maven_source_exists:
+            drivers.add('sourcedriver_extra_src_dir')
+            for maven_source_subdir in os.listdir(maven_source_data_dir):
+                cmd.append('-e')
+                cmd.append('{}'.format(os.path.join(maven_source_data_dir, maven_source_subdir)))
```

Comment:
I hope we won't reach limits here :D

On my system:
```
xargs --show-limits
Your environment variables take up 4465 bytes
POSIX upper limit on argument length (this system): 2090639
POSIX smallest allowable upper limit on argument length (all systems): 4096
Maximum length of command we could actually use: 2086174
Size of command buffer we are actually using: 131072
Maximum parallelism (--max-procs must be no greater): 2147483647
```

Reply:
It is definitely a possibility. A solution could be to file an RFE in the BSI script to iterate through the subdirectories on its own to avoid having to pass the full path of each and every subdirectory.

Reply:
Created https://github.com/containers/BuildSourceImage/issues/79